src/dataflow/src/sink/tail.rs

```diff
@@ -56,18 +68,26 @@ pub fn tail<G>(
                     while cursor.val_valid(&batch) {
                         let row = cursor.val(&batch);
                         cursor.map_times(&batch, |time, diff| {
-                            assert!(*diff >= 0, "negative multiplicities sinked in tail");
-                            let diff = *diff as usize;
+                            let diff = *diff;
                             let should_emit = if as_of.strict {
                                 as_of.frontier.less_than(time)
                             } else {
                                 as_of.frontier.less_equal(time)
                             };
                             if should_emit {
-                                for _ in 0..diff {
-                                    // Add the unpacked timestamp so we can sort by them later.
-                                    results.push((*time, row.clone()));
+                                packer.push(Datum::from(numeric::Numeric::from(*time)));
+                                if connector.emit_progress {
+                                    packer.push(Datum::False);
```

Comment:
What does this `Datum::False` indicate?

Reply:
It's about this one: https://materialize.com/docs/sql/tail/#progress

But you're right, I'll add an inline comment explaining this.